src/fluree/db/json_ld/transact.cljc

```diff
@@ -88,8 +88,13 @@
           flakes      (cond
                         ;; a new node's data is contained, process as another node then link to this one
                         (jld-reify/node? v-map)
-                        (let [[node-sid node-flakes] (<? (json-ld-node->flakes v-map tx-state pid))]
-                          (conj node-flakes (flake/create sid pid node-sid const/$xsd:anyURI t true m)))
+                        (do
+                          (when validate-fn
```

Comment:
Is there a reason to do the validation here instead of in `shacl/validate-shape` where we have all the other validation?

Reply:
> Is there a reason to do the validation here instead of in `shacl/validate-shape` where we have all the other validation?

If you look a little further down this is where `validate-fn`s get run on literal values, so it seemed like the right place for nodes too (short of a larger refactor).

Reply:
Yeah, I had always understood this `validate-fn` pattern to be an optimization for cases where we can reject invalid data really early, rather than waiting to run `validate-rules` after the `final-db` has been assembled.

Reply:
Nice, thanks. I've been spending a _lot_ of time in the shacl ns, I didn't realize there was a whole other area that did validation : )